.github/workflows/docker.yml

```diff
@@ -0,0 +1,81 @@
+name: Create and publish a Docker images
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+    branches:
+      - "main"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - php-version: '7.4'
+            php-suffix: '-php7.4'
+            alpine-version: '3.16'
+          - php-version: '8.0'
+            php-suffix: '-php8.0'
+            alpine-version: '3.16'
+          - php-version: '8.1'
+            php-suffix: '-php8.1'
+            alpine-version: '3.18'
+          - php-version: '8.2'
+            php-suffix: '-php8.2'
+            alpine-version: '3.18'
+          - php-version: '8.3'
+            php-suffix: '-php8.3'
+            alpine-version: '3.18'
+          - php-version: 'latest'
+            php-suffix: ''
+            alpine-version: '3.18'
```

Comment:
I'm unsure, if it's a good idea to drop this. We would exchange control over comfort, as the consequence would be losing all the handy tags like `php-cs-fixer:3`. On the other hand, the risk of "random" version is quite high, as you never know what you get for which tag.

Maybe i should leave it commented-out if we want to have it later?

Reply:
P.S. or do you mean taking 8.3 hardcoded instead of "latest"?

Reply:
The thing is Fixer has built-in conditions for supported PHP version, so even if 8.4 is out and we build such image, without changes in the code (or without setting `PHP_CS_FIXER_IGNORE_ENV` it won't work anyway). Such image could be helpful for testing purposes (verifying if it works), but the env variable should be set automatically (not sure about it, though) _OR_ must be documented in the usage docs.

Reply:
There is also risk that Docker build will fail because of Composer dependencies being not compatible with this PHP version (or we should use `--ignore-platform-req=php+` to make Composer ignore this requirement), so this particular job should be allowed to fail and should not block any other jobs and workflows.

Reply:
According to `composer.json` php-cs-fixer is compatible to PHP 8.x including all future versions. I think we should match to this contraint by either adjusting the composer.json to `<8.4` and use 8.3 for latest, or by setting the latest php image to `8`, so it should work the same as if I install it via composer in a php 8.4 installation

Reply:
We can't touch `composer.json`, because the constraint is done this way on purpose, to allow running on higher versions of PHP as an explicit opt-in. What is important in this case is [this](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/ac9c2e055f06204efbd37611e005142277bde937/php-cs-fixer#L31) - it's the `php-cs-fixer` that applies final range of supported PHP versions. If we change the constraint, then it would be impossible to install Fixer on 8.4+ and we don't want it. @keradus can you confirm?

PS. we _could_ change the constraint, but it would require using `--ignore-platform-req=php+` on the users' side when Fixer is installed, which is an option too, but I don't know iw @keradus wants to change the approach.

Reply:
That makes sense. So I will use 8.3 for the unsuffixed version.

Reply:
Continuation [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7555#pullrequestreview-1790644845).